mu/modes/microbit.py

```diff
@@ -291,34 +312,86 @@ def flash(self):
                              self.user_defined_microbit_path))
         # Check the path and that it exists simply because the path maybe based
         # on stale data.
-        logger.debug('Path to micro:bit: {}'.format(path_to_microbit))
         if path_to_microbit and os.path.exists(path_to_microbit):
-            logger.debug('Flashing to device.')
+            force_flash = False  # If set to true, fully flash the device.
+            logger.info("Checking target device.")
+            # Get the version of MicroPython on the device.
+            try:
+                version_info = microfs.version()
+                logger.info(version_info)
+                board_version = version_info['version'].split('-', 1)[0]
+                board_version = board_version.replace('micro:bit v', '')
+                logger.info('Board MicroPython: {}'.format(board_version))
```

Comment:
Yeah... I noticed this a few days ago... I'll revise extraction of the version number in a moment.

Reply:
OK, for this to work I need to know that `b'e10a5ff'` isn't just a byte representation problem because someone somewhere used Python 2 in the build process. :-/

Assuming semver, then I guess we're looking for version numbers that look like this: `1.0-beta.1+e10a5ff` which it currently doesn't look like.

Reply:
Yes, you are right, we'll update this to follow semver.

Reply:
About the b'e10a5ff', I am not 100% sure yet when it is tagged like that. I've used yotta on Python 2 for my builds and it **doesn't** append as `b'e10a5ff'`, it's possible this happens when using Python 3, as that could be used in Damien's environment.

We do need to get to the bottom of this, but for the time being, this would be metadata ignored by the semver parser comparison, no?

Reply:
Yeah... so long as it follows a "+" in the version string, as per https://semver.org/#spec-item-10

There's a Python module called `semver` which, if our versions are correct, will handle all this correctly. This (unfortunately) adds a new dependency at this late stage, which makes me feel very nervous. Happily it's already packaged for Debian, no idea about Fedora.